src/goto-symex/renaming_level.h

```diff
@@ -84,6 +86,63 @@ struct symex_level2t : public symex_renaming_levelt
 {
   symex_level2t() = default;
   ~symex_level2t() override = default;
+
+  /// In the case of a state split, we want to keep track of the overall
+  /// name generation across all states for identifier creation.
+  optionalt<current_namest> local_generations;
+
+  void erase(const irep_idt &identifier)
+  {
+    current_names.erase(identifier);
```

Comment:
Given `current_names` is the L2 global unique generation, and the `erase` method is path-local, I think this is dubious.